src/GeneratorState.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef GENSET_WHISPERPOWER_DDC_GENERATOR_STATE_HPP
+#define GENSET_WHISPERPOWER_DDC_GENERATOR_STATE_HPP
+
+#include <base/Time.hpp>
+#include <genset_whisperpower_ddc/GeneratorStatus.hpp>
+
+namespace genset_whisperpower_ddc {
+    struct GeneratorState {
+        base::Time time;
+
+        int rpm;
+        int udc_start_battery;
+        enum Status {
+            OVERALL_ALARM = 0x01,
+            ENGINE_TEMPERATURE_ALARM = 0x02,
+            PM_VOLTAGE_ALARM = 0x04,
+            OIL_PRESSURE_ALARM = 0x08,
+            EXHAUST_TEMPERATURE_ALARM = 0x10,
+            UAC1_ALARM = 0x20,
+            IAC1_ALARM = 0x40,
+            OIL_PRESSURE_HIGH_ALARM = 0x80,
+            LOW_START_BATTERY_VOLTAGE_ALARM = 0x0400,
+            START_FAILURE = 0x1000,
+            RUN_SIGNAL = 0x2000,
+            START_BY_OPERATION_UNIT = 0x8000,
+            MODEL_DETECTION_50HZ = 0x040000,
+            MODEL_DETECTION_60HZ = 0x080000,
+            MODEL_DETECTION_3_PHASE = 0x100000,
+            MODEL_DETECTION_MOBILE = 0x200000
+        };
+
+        /** @meta bitfield /genset_whisperpower_ddc/GeneratorState/Status */
+        uint32_t status = 0;
+
+        GeneratorStatus generator_status;
+        int generator_type;
```

Comment:
Avoid repeatedly sending an information that, essentially, you already know.

This part, as well as the MODEL_DETECTION bits would preferably be stored in their own data structure. You can then return them separately from the same API method. The idea would be to have only the (dynamic) status on a component port.

Reply:
Are you referring to both generator_status or just generator_type? Can't generator_status be considerated dynamic?

Reply:
Does that mean that the GeneratorModel structure shouldn't be output by the component or that it should but in another port? Btw, just to be sure, by "this part" you meant just the generator_type or also generator_status? Because generator_status gives new information, right?

Reply:
The only thing lefty in `generator_status` are MODEL_DETECTION_50HZ, MODEL_DETECTION_60HZ, MODEL_DETECTION_3_PHASE and MODEL_DETECTION_MOBILE

Which are not needed.

No, they will not be output by the component. At best, we could save them in an attribute, but honestly there is zero value IMO. The only place where they could be useful is in the command-line interface.

Reply:
Ah ... was referring to the model bits from `status` (from the original PR). `generator_status` stays.